tests/colr_to_svg_test.py

```diff
@@ -0,0 +1,145 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from fontTools.colorLib.builder import buildCPAL, buildColrV1, LayerListBuilder
+from fontTools import ttLib
+from fontTools.ttLib.tables import _g_l_y_f
+from fontTools.ttLib.ttFont import newTable
+from fontTools.pens.ttGlyphPen import TTGlyphPen
+from fontTools.ttLib.tables import otTables as ot
+from lxml import etree
+from nanoemoji.colr_to_svg import _colr_v1_paint_to_svg, _new_reuse_cache
+from nanoemoji.util import only
+from picosvg.svg import SVG
+from picosvg.svg_transform import Affine2D
+import pytest
+from test_helper import svg_diff
+import textwrap
+
+
+def _draw_box(pen):
+    pen.moveTo((10, 10))
+    pen.lineTo((90, 10))
+    pen.lineTo((90, 90))
+    pen.lineTo((10, 90))
+    pen.closePath()
+
+
+def _build(cls, source) -> ot.Paint:
+    return LayerListBuilder().tableBuilder.build(cls, source)
+
+
+def _buildPaint(source) -> ot.Paint:
+    return LayerListBuilder().buildPaint(source)
+
+
+@pytest.mark.parametrize(
+    "glyph_to_convert, color_glyphs, monochrome_glyphs, expected_svg",
+    [
+        # Solid filled box
+        (
+            "color_box",
+            {
+                "color_box": (
+                    ot.PaintFormat.PaintGlyph,
+                    (ot.PaintFormat.PaintSolid, 0),
+                    "box",
+                ),
+            },
+            {"box": _draw_box},
+            """
+            <svg xmlns="http://www.w3.org/2000/svg">
+              <defs/>
+              <path d="M10,10 L90,10 L90,90 L10,90 Z"/>
+            </svg>
+            """,
+        ),
+        # Paint colr glyph
+        (
+            "paint_colr_glyph",
+            {
+                "paint_colr_glyph": (ot.PaintFormat.PaintColrGlyph, "color_box"),
```

Comment:
an even more interesting test case would be a PaintColrGlyph inside a PaintTransform -- which, incidentally, is what glyhpsLib produces for composite color glyphs found in .glyphs3 sources (cf. ArefRuqaInk)